### PR TITLE
Virtual applications docs #441

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

File: docs/cms/sites.adoc [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 A highly central component of the XP CMS are sites.
 Site is shipped as a standard content type, which means it can easily be created and added by editors.
-However, sites have the unique ability to let you add applications to them.
+However, sites have an ability to let you add applications to them. Site applications isolated from <<projects#, projects>> applications and can't be inherited.
```
Comment (rymsha) [inline docs/cms/sites.adoc:9]:
```suggestion
However, sites have an ability to let you add applications to them. Site applications isolated from <<projects#, projects>> applications. If a site needs an application from its project - add the application to the site.
```

**Review by rymsha — CHANGES_REQUESTED** [review]

File: docs/apps.adoc [hunk 2]
```diff
@@ -58,6 +62,12 @@ Projects contain all code and configuration required to build the application.
 
 For more details, check out the <<apps/projects#, project reference documentation>>.
 
+== Virtual Applications
+
+image:xp-7110.svg[XP 7.11.0,opts=inline]
+
+For more details, check out the <<apps/virtual-applications#, virtual applications>>.
```
Comment (alansemenov) [inline docs/apps.adoc:69]:
```suggestion
For more details, check out the <<apps/virtual-applications#, virtual applications documentation>>.
```

File: docs/apps/projects.adoc [hunk 3]
```diff
@@ -8,7 +8,7 @@ How are applications and libraries put together built and delivered
 
 The XP project structure is based on the https://maven.apache.org/[Maven] project structure.
 
-Below is is an extensive version of a project structure, listing all standard folders and files that can be used
+Below it is an extensive version of a project structure, listing all standard folders and files that can be used
```
Comment (alansemenov) [inline docs/apps/projects.adoc:11]:
```suggestion
Below is an extensive version of a project structure, listing all standard folders and files that can be used
```

File: docs/apps/virtual-applications.adoc [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      parts/
+      x-data/
+      site.xml
+      styles.xml
+
+----
+Use <<../api/lib-schema, lib-schema>> to manipulate schemas and descriptors of virtual apps.
+
+=== Application Mode
+Depends on app config, xp can work with application resources in different modes:
```
Comment (alansemenov) [inline docs/apps/virtual-applications.adoc:29]:
```suggestion
Depending on app config, XP can work with application resources in different modes:
```

File: docs/apps/virtual-applications.adoc [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+VIRTUAL::
+    If specified resource is presented in both virtual and bundled app, then virtual will be used.
```
Comment (alansemenov) [inline docs/apps/virtual-applications.adoc:32]:
```suggestion
    If a specified resource is present in both a virtual and a bundled app, then the virtual one will be used.
```

File: docs/apps/virtual-applications.adoc [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+BUNDLED::
+    Resources only from bundled applications are used, virtual apps are disabled and its resources are ignored.
```
Comment (alansemenov) [inline docs/apps/virtual-applications.adoc:34]:
```suggestion
    Only resources from bundled applications are used, virtual apps are disabled and their resources are ignored.
```

File: docs/apps/virtual-applications.adoc [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+AUGMENTED::
+    If specified resource is presented in both virtual and bundled app, then virtual will be used.
```
Comment (alansemenov) [inline docs/apps/virtual-applications.adoc:36]: This is exactly the same as description for "VIRTUAL", and doesn't seem right.

File: docs/apps/virtual-applications.adoc [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+
+Check the <<../deployment/config#application, documentation>> to find how config the mode.
```
Comment (alansemenov) [inline docs/apps/virtual-applications.adoc:39]:
```suggestion
Check the <<../deployment/config#application, documentation>> to find out how to configure each mode.
```

File: docs/cms/projects.adoc [hunk 9]
```diff
@@ -39,6 +38,14 @@ For each project, dedicated principal roles are created. The roles are named as
 
 An example might be com.enonic.cms.myproject.author.
 
+=== Applications
+image:xp-7110.svg[XP 7.11.0,opts=inline]
+
+Just like with sites, applications can be added to projects. All functionality and all schemas (content types, schemas, mixins) and descriptors (pages, parts, layouts) from the app will be available within the scope of the project down to the site. Sites override project applications. If a site needs an application from its project - add the application to the site.
```
Comment (alansemenov) [inline docs/cms/projects.adoc:44]:
```suggestion
Just like with sites, applications can be added to projects. All functionality and all schemas (content types, schemas, mixins) and descriptors (pages, parts, layouts) from an app will be available within the scope of the project down to a site. 

IMPORTANT: Sites override project applications. If a site should use an application already assigned to its project, you must assign the application directly to the site.
```

File: docs/cms/sites.adoc [hunk 10]
```diff
@@ -6,7 +6,7 @@
 
 A highly central component of the XP CMS are sites.
 Site is shipped as a standard content type, which means it can easily be created and added by editors.
-However, sites have the unique ability to let you add applications to them.
+However, sites have an ability to let you add applications to them. Site applications isolated from <<projects#, projects>> applications. If a site needs an application from its project - add the application to the site.
```
Comment (alansemenov) [inline docs/cms/sites.adoc:9]:
```suggestion
Sites can also have applications assigned to them. Site applications are isolated from <<projects#, project>> applications. If a site should use an application already assigned to its project, you must assign the application directly to the site.
```

File: docs/apps/virtual-applications.adoc [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+Shows that only virtual application is presented in the system and support of virtual apps is enabled.
+BUNDLED::
+Shows one of the next cases:
+- Only bundled application is presented in the system.
+- Both bundled and virtual applications are presented, but bundled resources override virtual.
+- Both bundled and virtual applications are presented, but virtual app support is disabled.
+AUGMENTED::
+Both bundled and virtual applications are presented in the system and virtual resources override bundled.
+
+Check the <<../deployment/config#application, documentation>> to find out how to configure application behaviour.
```
Comment (alansemenov) [inline docs/apps/virtual-applications.adoc:41]:
```suggestion
VIRTUAL::
Shows that only virtual application is present in the system and support of virtual apps is enabled.
BUNDLED::
Can be one of the next cases:
- Only bundled application is present in the system.
- Both bundled and virtual applications are present, but bundled resources are configured to override virtual.
- Both bundled and virtual applications are present, but virtual app support is disabled via config.
AUGMENTED::
Both bundled and virtual applications are present in the system and virtual resources override bundled.

Check the <<../deployment/config#application, documentation>> to find out how to configure application behaviour.
```

**Review by alansemenov — APPROVED** [review]

**Review by rymsha — APPROVED** [review]

**Pull request merged**